server/middleware/auth.js

```diff
@@ -0,0 +1,43 @@
+const jwt = require('jsonwebtoken');
+
+exports.loginVerify = async function (req, res, next) {
+      try {
+            const token = req.headers.authorization.split(' ')[1];
+            const decoded = jwt.verify(token, process.env.JWT_SECRET_KEY);
+            if (decoded) return next();
+            else return next({
+                  status: 401,
+                  message: 'Please sign in!'
+            });
+      } catch (err) {
+            return next(err);
+      }
+}
+
+exports.userVerify = async function (req, res, next) {
+      try {
+            const token = req.headers.authorization.split(' ')[1];
+            const decoded = jwt.verify(token, process.env.JWT_SECRET_KEY);
+            if (decoded && decoded.id === req.params.id) return next();
+            else return next({
+                  status: 401,
+                  message: 'Unauthorized'
+            })
+      } catch (err) {
+            return next(err);
+      }
+}
+
+exports.vendorVerify = async function (req, res, next) {
+      try {
+            const token = req.headers.authorization.split(' ')[1];
+            const decoded = jwt.verify(token, process.env.JWT_SECRET_KEY);
+            if (decoded && decoded.category === 'VENDOR') return next();
```

Comment:
where is the decoded.category defined

Reply:
category 'VENDOR' is defined in ``/models/user.js``. In the pre-hook:
```
// vendor category
userSchema.pre('save', async function (next) {
      try {
            if (this.email.includes('@vendor.com')) this.category = 'VENDOR';
            return next();
      } catch (err) {
            return next(err)
      }
});
```
And I wrap it into the token with ``jwt.sign`` in ``/handlers/auth.js``. The token will be saved in the ``localStorage`` of the React and then will be used in ``decoded`` to check if this account is vendor or user.